articles/build-a-jekyll-site/index.md

```diff
@@ -0,0 +1,358 @@
+
+```bash
+$ echo 'export GEM_HOME="$HOME/gems"' >> ~/.bashrc
+```
+
+```bash
+$ echo 'export PATH="$HOME/gems/bin:$PATH"' >> ~/.bashrc
+```
+
+Then execute the `.bashrc` file.
+
+```bash
+$ source ~/.bashrc
+```
+
+### Step 2 -- Install Jekyll
+
+Jekyll comes as a [Ruby Gem](https://guides.rubygems.org/what-is-a-gem/) and can be installed in any system running Ruby. Run the command below in a terminal to install Jekyll.
+
+```bash
+$ gem install jekyll bundler
+```
+
+### Step 3 -- Create a new Jekyll site
+Create a new folder where you will be creating your blog.
+
+```bash
+$ mkdir blog && cd blog
+```
+
+Create a new Jekyll site using the command below.
+
+```bash
+$ jekyll new blog
+```
+
+`blog` is the site name. Feel free to change the site name. The command creates files with the following folder structure.
+
+```
+├── 404.html
+├── about.markdown
+├── _config.yml
+├── Gemfile
+├── Gemfile.lock
+├── index.markdown
+├── _posts
+│   └── 2020-09-14-welcome-to-jekyll.markdown
+└── _site
+```
+
+We'll look at this structure in depth later. Let's run the site now. Run the following command in a terminal.
+
+```bash
+$ bundle exec jekyll serve
+```
+
+The command builds your site and you can access the site at http://127.0.0.1:4000 by default. You can run the site using the `--livereload` flag to auto-reload the site after making changes to it.
+
+```bash
+$ bundle exec jekyll serve --livereload
+```
+
+### Understanding the Folder Structure
+Now, let's look at the site structure. There are 2 main folders, `_site` and `_posts`. The `_posts` folder contains blog posts while `_site` contains the generated site. You should not edit the contents of this folder.
+
+The `_config.yml`  is a  YAML file that stores site settings and **rarely** changed variables. Go ahead and edit some site properties available in this file. For example, name, description, etc.
+
+The next important file is the `Gemfile`. The site dependencies are defined here. There is also a `Gemfile.lock` file which is a more detailed version of the `Gemfile`. The specific versions of the **currently installed** dependencies defined in the Gemfile are recorded in this file. This file is important as it prevents version conflicts if you were to share your site as a theme. This file shouldn't be edited directly.
+
+You can learn more about the folder structure [here](https://jekyllrb.com/docs/structure/).
+
+### Step 4 -- Add blog posts
+Open the main folder in your favorite code editor and navigate to the `_posts` folder. There is a welcome post written in markdown. The blog posts file names follow a `date-slug.md` naming convention. The date should be formatted like this `YYYY-MM-DD`. The slug is the part of the URL identifying a particular post.
+
+To create a new post, create a markdown file and name it `2020-09-14-my-new-site.md`. Feel free to change the date and slug. If you reload your site, you can see you have a new post. If you click it at this point, it opens a blank page.
```

Comment:
It would be beneficial to explain what Markdown is and add a brief section explaining common formatting (or link to a Markdown guide.)